source/material_model/reactive_fluid_transport.cc

```diff
@@ -50,6 +50,96 @@ namespace aspect
 
 
 
+    template <int dim>
+    std::vector<double>
+    ReactiveFluidTransport<dim>::
+    tian_equilibrium_bound_water_content (const MaterialModel::MaterialModelInputs<dim> &in,
+                                          unsigned int q) const
+    {
+      // Pressure, which must be in GPa for the parametrization, or GPa^-1
+      const double pressure = in.pressure[q]<=0 ? 1e-12 : in.pressure[q]/1e9;
+      const double inverse_pressure = std::pow(pressure, -1);
+
+      // The following coefficients are taken from a publication from Tian et al., 2019, and can be found
+      // in Table 3 (Gabbro), Table B1 (MORB), Table B2 (Sediments) and Table B3 (peridotite).
+      // LR refers to the effective enthalpy change for devolatilization reactions,
+      // csat is the saturated mass fraction of water in the solid, and Td is the
+      // onset temperature of devolatilization for water.
+      std::vector<double> LR_peridotite_poly_coeffs {-19.0609, 168.983, -630.032, 1281.84, -1543.14, 1111.88, -459.142, 95.4143, 1.97246};
+      std::vector<double> csat_peridotite_poly_coeffs {0.00115628, 2.42179};
+      std::vector<double> Td_peridotite_poly_coeffs {-15.4627, 94.9716, 636.603};
+
+      std::vector<double> LR_gabbro_poly_coeffs {-1.81745, 7.67198, -10.8507, 5.09329, 8.14519};
+      std::vector<double> csat_gabbro_poly_coeffs {-0.0176673, 0.0893044, 1.52732};
+      std::vector<double> Td_gabbro_poly_coeffs {-1.72277, 20.5898, 637.517};
+
+      std::vector<double> LR_MORB_poly_coeffs {-1.78177, 7.50871, -10.4840, 5.19725, 7.96365};
+      std::vector<double> csat_MORB_poly_coeffs {0.0102725, -0.115390, 0.324452, 1.41588};
+      std::vector<double> Td_MORB_poly_coeffs {-3.81280, 22.7809, 638.049};
+
+      std::vector<double> LR_sediment_poly_coeffs {-2.03283, 10.8186, -21.2119, 18.3351, -6.48711, 8.32459};
+      std::vector<double> csat_sediment_poly_coeffs {-0.150662, 0.301807, 1.01867};
+      std::vector<double> Td_sediment_poly_coeffs {2.83277, -24.7593, 85.9090, 524.898};
+
+      std::vector<std::vector<double>> LR_all_poly_coeffs {LR_peridotite_poly_coeffs, LR_gabbro_poly_coeffs, \
+                                                            LR_MORB_poly_coeffs, LR_sediment_poly_coeffs
+                                                           };
+
+      std::vector<std::vector<double>> csat_all_poly_coeffs {csat_peridotite_poly_coeffs, csat_gabbro_poly_coeffs, \
+                                                              csat_MORB_poly_coeffs, csat_sediment_poly_coeffs
+                                                             };
+
+      std::vector<std::vector<double>> Td_all_poly_coeffs {Td_peridotite_poly_coeffs, Td_gabbro_poly_coeffs, \
+                                                            Td_MORB_poly_coeffs, Td_sediment_poly_coeffs
+                                                           };
```

Comment:
Is there a  reason you create these vectors here rather than in the .h file? This means that every time you call the function (in other words, for each cell in the model), they will be created and destroyed again.